google-cloud-storage/acceptance/storage/bucket_post_object_v4_test.rb

```diff
@@ -0,0 +1,135 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "storage_helper"
+require 'net/http'
+require 'uri'
+
+describe Google::Cloud::Storage::Bucket, :post_object_v4, :storage do
+  let(:bucket_name) { $bucket_names.first }
+  let :bucket do
+    storage.bucket(bucket_name) ||
+    safe_gcs_execute { storage.create_bucket bucket_name }
+  end
+  let(:uri) { URI.parse Google::Cloud::Storage::GOOGLEAPIS_URL }
+  let(:data_file) { "logo.jpg" }
+  let(:data) { File.expand_path("../data/#{data_file}", __dir__) }
+
+  it "generates a signed post object v4 simple" do
+    post_object = bucket.post_object_v4 "test-object", expires: 10
+
+    form_data = [
+      ["file", File.open(data)],
+      ["key", post_object.fields["key"]],
+      ["x-goog-algorithm", post_object.fields["x-goog-algorithm"]],
+      ["x-goog-credential", post_object.fields["x-goog-credential"]],
+      ["x-goog-date", post_object.fields["x-goog-date"]],
+      ["x-goog-signature", post_object.fields["x-goog-signature"]],
+      ["policy", post_object.fields["policy"]]
```

Comment:
could you do me a favor and rewrite this with:
```ruby
form_data = [['file', File.open(data)]]

post_object.fields.each do |key, value|
  form_data.push([key, value])
end
```

To rely on implementation providing users with the exact information needed to make a successful request.

Reply:
Updated.

Reply:
Thank you, this issue is resolved.